base/doc/clsguide.tex

```diff
@@ -527,54 +353,43 @@ \subsection{Identification}
 For example:
 \begin{verbatim}
    \NeedsTeXFormat{LaTeX2e}
-   \ProvidesClass{article}[1994/06/01 Standard LaTeX class]
+   \ProvidesClass{article}[2002-06-01 Standard LaTeX class]
 \end{verbatim}
-\NEWdescription{1998/06/19}
-The \m{date} should be given in the form `\textsc{yyyy/mm/dd}' and
-must be present if the optional argument is used (this is also true
-for the |\NeedsTeXFormat| command).
-Any derivation from this syntax will result in low-level \TeX{}
-errors---the commands expect a valid syntax to speed up the daily
-usage of the package or class and make no provision for the case that
-the developer made a mistake!
-
-This date is checked whenever a user specifies a date in their
-|\documentclass| or |\usepackage| command.  For example, if you wrote:
+The \m{date} should be given in the form `\textsc{yyyy-mm-dd}' and must be
+present if the optional argument is used (this is also true for the
+|\NeedsTeXFormat| command). Any derivation from this syntax will result in
+low-level \TeX{} errors---the commands expect a valid syntax to speed up the
+daily usage of the package or class and make no provision for the case that the
+developer made a mistake!
+
+This date is checked whenever a user specifies a date in their |\documentclass|
+or |\usepackage| command. For example, if you wrote:
 \begin{verbatim}
-   \documentclass{article}[1995/12/23]
+   \documentclass{article}[2002-06-01]
 \end{verbatim}
-then users at a different location
-would get a warning that their copy of |article| was out of
-date.
-
-The description of a class is displayed when the class is used.  The
-description of a package is put into the log file.  These descriptions
-are also displayed by the |\listfiles| command.  The phrase
-\texttt{Standard LaTeX} \textbf{must not} be used in the identification
-banner of any file other than those in the standard \LaTeX{}
-distribution.
+then users at a different location would get a warning that their copy of
+|article| was out of date.
 
+The description of a class is displayed when the class is used. The description
+of a package is put into the log file. These descriptions are also displayed by
+the |\listfiles| command. The phrase \texttt{Standard LaTeX} \textbf{must not}
+be used in the identification banner of any file other than those in the
+standard \LaTeX{} distribution.
 
 \subsection{Using classes and packages}
 
-The first major difference between \LaTeX~2.09 style files and
-\LaTeXe{} packages and classes is that \LaTeXe{} supports
-\emph{modularity}, in the sense of building files from small
-building-blocks rather than using large single files.
-
 A \LaTeX{} package or class can load a package as follows:
 \begin{verbatim}
    \RequirePackage[<options>]{<package>}[<date>]
 \end{verbatim}
```

Comment:
Maybe something along the lines:

The use of `[=<date>]`  or `[=<version>]`  in the second optional argument should be avoided in new classes or packages. This is really intended as a rollback possibility for the user, i.e., for use in the preamble. Up-to-date packages or classes should not force old/obsolete versions of other packages because that generates unresolvable and surprising dependencies. Of course, if you provide rollback information for your own package, then that code might require to also roll back other packages to the same date.